pytorch_lightning/trainer/trainer.py

```diff
@@ -941,7 +942,7 @@ def fit(
 
         # set up the passed in dataloaders (if needed)
         self.__attach_dataloaders(model, train_dataloader, val_dataloaders)
-        self.__attach_datamodule(model, datamodule)
+        self.__attach_datamodule(model, datamodule, 'fit')
```

Comment:
```suggestion
        if not self.testing:
            self.__attach_datamodule(model, datamodule, 'fit')
```

Reply:
Ah this is more in line w/ everything else. nice

Reply:
Also cover the `__attach_dataloaders` under `if not self.testing` condition too I think.